core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -0,0 +1,137 @@
+import lombok.ToString;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.rnorth.ducttape.TimeoutException;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.output.FrameConsumerResultCallback;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * 'Sidekick container' with the sole purpose of recording the VNC screen output from another container.
+ *
+ */
+@Getter
+@ToString
+public class VncRecordingContainer extends GenericContainer<VncRecordingContainer> {
+
+    private static final String RECORDING_FILE_NAME = "/screen.flv";
+
+    public static final String DEFAULT_VNC_PASSWORD = "secret";
+
+    public static final int DEFAULT_VNC_PORT = 5900;
+
+    private final String targetNetworkAlias;
+
+    private String vncPassword = DEFAULT_VNC_PASSWORD;
+
+    private int vncPort = 5900;
+
+    private int frameRate = 30;
+
+    public VncRecordingContainer(@NonNull GenericContainer<?> targetContainer) throws IllegalStateException {
+        this(
+                targetContainer.getNetwork(),
+                targetContainer.getNetworkAliases().stream()
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("Target container must have a network alias"))
+        );
+    }
+
+    /**
+     * Create a sidekick container and attach it to another container. The VNC output of that container will be recorded.
+     */
+    public VncRecordingContainer(@NonNull Network network, @NonNull String targetNetworkAlias) throws IllegalStateException {
+        super(TestcontainersConfiguration.getInstance().getVncRecordedContainerImage());
+
+        this.targetNetworkAlias = targetNetworkAlias;
+        withNetwork(network);
+
+        waitingFor(new AbstractWaitStrategy() {
+
+            @Override
+            protected void waitUntilReady() {
+                try {
+                    Unreliables.retryUntilTrue((int) startupTimeout.toMillis(), TimeUnit.MILLISECONDS, () -> {
+                        CountDownLatch latch = new CountDownLatch(1);
+
+                        FrameConsumerResultCallback callback = new FrameConsumerResultCallback() {
+                            @Override
+                            public void onNext(Frame frame) {
+                                if (frame != null && new String(frame.getPayload()).contains("Connected")) {
+                                    latch.countDown();
+                                }
+                            }
+                        };
+
+                        try (
+                                Closeable __ = dockerClient.logContainerCmd(containerId)
```

Comment:
`__` == unused

Reply:
Starting with Java 9, underscores won't be legal variable names ([Java Language Changes for Java SE 9](https://docs.oracle.com/javase/9/language/toc.htm#JSLAN-GUID-E409CC44-9A8F-4043-82C8-6B95CD939296)) (okay, maybe double underscore is fine 😜 )